### ISSUE TITLE
loss is NAN or become larger

### ISSUE THREAD

**Issue by DongHappyyy** [luminxu/ViPNAS#8]
Hello, I have a problem during training. I use 3090 single card training, but loss will become Nan or suddenly become larger

**Comment by DongHappyyy** [comment]
[20211005_125611.log](https://github.com/luminxu/ViPNAS/files/7303662/20211005_125611.log)


I have tried to modify the learning rate and increase the batch. My learning rate is changed to 1e-10 and still Nan

**Comment by luminxu** [comment]
I try to train [S-ViPNAS](https://github.com/luminxu/ViPNAS/blob/main/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/s_vipnas_res50_coco_256x192.py) using a single GPU (V100), which achieves comparable performance throughout the whole training process and does not face the problem. I notice that your training loss is different from [ours](https://drive.google.com/file/d/1VRmsvPKvpaPqHqVZYqmX1aatGzfmCWRS/view) at the very beginning, and the loss almost does not decrease from the 2nd epoch.

I suggest 1) Please make sure that you have correctly installed the environment following the [instruction](https://github.com/luminxu/ViPNAS/blob/main/README.md).  2) Train other models provided in [MMPose](https://github.com/open-mmlab/mmpose) to check whether the same problem happens.

**Comment by DongHappyyy** [comment]
Can you tell me your environment configuration, such as CUDA version, torch version and so on. Thanks !

**Comment by luminxu** [comment]
I use CUDA V11.0, and PyTorch 1.3+ would be fine. You can follow [ViPNAS](https://github.com/luminxu/ViPNAS/blob/main/README.md) or [MMPose](https://github.com/open-mmlab/mmpose/blob/master/docs/install.md) to install your environment.

**Comment by DongHappyyy** [comment]
Thank you very much. I try it now!

**Comment by DongHappyyy** [comment]
I have solved this problem, not because of the code, but because of my machine. I changed a device and it works normally。
Thank you very much!

**Comment by DongHappyyy** [comment]
But now I have a new question, which may be relatively low-level, but I am still eager for your answer. Now I want to test my own pictures, input a picture into the program, and output the key points I want. But I don't have the “center" and "scale" of img_metas parameters now. How should I test my own pictures?

**Comment by luminxu** [comment]
As ViPNAS follows the top-down paradigm, human bounding boxes are required for multi-person pose estimation. For an input image, the common solution is to apply a human detector for bounding boxes then the pose estimator for keypoints.  You can use [top_down_img_demo_with_mmdet.py](https://github.com/luminxu/ViPNAS/blob/main/demo/top_down_img_demo_with_mmdet.py) to test on your own images, which apply the human detector provided by [MMDetection](https://github.com/open-mmlab/mmdetection).  Please refer to [2D Human Pose Demo](https://github.com/luminxu/ViPNAS/blob/main/demo/docs/2d_human_pose_demo.md) for more information.

As the problem of training loss has been addressed, I will close the issue.

**Comment by DongHappyyy** [comment]
Thank you！ Thank you very much!  Thank you very much for your answer, which has given me great help！ Thank you again！

**Comment by luminxu** [comment]
You are welcome : ) Hope you can enjoy our work.
